apps/web/src/components/Posts/PostComposerModal.tsx

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import ErrorBoundary from '~/contexts/boundary/ErrorBoundary';
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { PostComposerFragment$key } from '~/generated/PostComposerFragment.graphql';
+import { PostComposerModalFragment$key } from '~/generated/PostComposerModalFragment.graphql';
+import { PostComposerModalWithSelectorFragment$key } from '~/generated/PostComposerModalWithSelectorFragment.graphql';
+import { PostComposerModalWithSelectorQueryFragment$key } from '~/generated/PostComposerModalWithSelectorQueryFragment.graphql';
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+
+import breakpoints from '../core/breakpoints';
+import { Button } from '../core/Button/Button';
+import { VStack } from '../core/Spacer/Stack';
+import { BaseM } from '../core/Text/Text';
+import { NftSelector } from '../NftSelector/NftSelector';
+import PostComposer from './PostComposer';
+
+type Props = {
+  tokensRef: PostComposerModalWithSelectorFragment$key;
+  queryRef: PostComposerModalWithSelectorQueryFragment$key;
+};
+
+// Modal with multiple steps: the NFT Selector -> then Post Composer
+export function PostComposerModalWithSelector({ tokensRef, queryRef }: Props) {
```

Comment:
nice!! we could prob consolidate into a single component with an option (`withSelector: boolean`) but i think this is cleaner / easier to reason about, esp if the modals end up having diverging use cases

Reply:
yeahh exactly - if we find we're repeating code a bunch we can definitely consolidate. I was deep trying to separate the nft selector for pfp and post use cases so it also felt clean to separate these two modal flows